Touch Bar Simulator/AppDelegate.swift

```diff
@@ -75,19 +60,148 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
 	}
 
 	@objc
-	func setTransparency(sender: ToolbarSlider) {
-		window.alphaValue = CGFloat(sender.doubleValue)
-		defaults.set(sender.doubleValue, forKey: "windowTransparency")
+	func toggleView(_ pboard: NSPasteboard, userData: String, error: NSErrorPointer) {
+		toggleView()
 	}
 
-	@objc
-	func toggleView(_ pboard: NSPasteboard, userData: String, error: NSErrorPointer) {
+	func toggleView() {
 		window.setIsVisible(!window.isVisible)
 	}
+
+	private var statusMenuItemFloating: NSMenuItem!
+	private var statusMenuItemDockedToTop: NSMenuItem!
+	private var statusMenuItemDockedToBottom: NSMenuItem!
+	private var statusMenuDockingItems: [NSMenuItem]!
+	private var statusMenuItemShowOnAllDesktops: NSMenuItem!
+	private var statusMenuItems: [NSMenuItem]!
+
+	var docking: TouchBarWindow.Docking = .floating {
+		didSet {
+			defaults[.windowDocking] = docking
+
+			statusMenuDockingItems.forEach { $0.state = .off }
+
+			let onItem: NSMenuItem
+			switch docking {
+			case .floating:
+				onItem = statusMenuItemFloating
+			case .dockedToTop:
+				onItem = statusMenuItemDockedToTop
+			case .dockedToBottom:
+				onItem = statusMenuItemDockedToBottom
+			}
+			onItem.state = .on
```

Comment:
If https://github.com/sindresorhus/touch-bar-simulator/pull/33#discussion_r251212488 is done, this logic could be moved to the menu creation function.

Reply:
And with that done, you could probably just move all the docking logic to `TouchBarWindow.Docking`.